smart_open/s3.py

```diff
@@ -1,17 +1,29 @@
 # -*- coding: utf-8 -*-
 """Implements file-like objects for reading and writing from/to S3."""
-import boto3
-import botocore.client
 
 import io
+import contextlib
+import functools
+import itertools
 import logging
 
+import boto3
+import botocore.client
 import six
 
 
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
+# Multiprocessing is unavailable in App Engine (and possibly other sandboxes).
+# The only method currently relying on it is s3_iter_bucket, which is instructed
+# whether to use it by the MULTIPROCESSING flag.
+_MULTIPROCESSING = False
+try:
+    import multiprocessing.pool
+    _MULTIPROCESSING = True
```

Comment:
why can `multiprocessing` be unavailable?

Reply:
I trust the above comment. It may be unavailable in certain environments, but I've never been in one.